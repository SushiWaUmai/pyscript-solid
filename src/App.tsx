import PyEnv from "./components/PyEnv";
import PyScript from "./components/PyScript";

function App() {
  return (
    <div>
      <PyEnv>- numpy</PyEnv>
      <PyEnv>- pandas</PyEnv>
      <PyScript>print("Hello World")</PyScript>
    </div>
  );
}

export default App;
