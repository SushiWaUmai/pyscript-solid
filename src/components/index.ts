export * from "./PyBox";
export * from "./PyButton";
export * from "./PyConfig";
export * from "./PyEnv";
export * from "./PyInputBox";
export * from "./PyRegisterWidget";
export * from "./PyRepl";
export * from "./PyScript";
export * from "./PyTitle";
