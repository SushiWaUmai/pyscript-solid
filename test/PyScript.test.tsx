import { describe, expect, test } from "vitest";
import { render } from "solid-testing-library";
import { PyScript } from "../src";

describe("<PyScript />", () => {
  test("renders", () => {
    const { unmount, container } = render(() => <PyScript />);
    expect(container.hasAttribute("src")).toBe(false);
    unmount();
  });

  test("has attributes", () => {
    const { unmount, container } = render(() => <PyScript src="test.py" />);
    expect(container.getAttribute("src")).toBe("test.py");
    unmount();
  });

  test("has children", () => {
    const { unmount, container } = render(() => (
      <PyScript>print("Hello World!")</PyScript>
    ));
    expect(container.textContent).toBe(`print("Hello World!")`);
    unmount();
  });
});
