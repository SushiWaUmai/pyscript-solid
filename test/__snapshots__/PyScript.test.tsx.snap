// Vitest Snapshot v1

exports[`<PyScript /> > has attributes 1`] = `"<py-script src=\\"test.py\\"></py-script>"`;

exports[`<PyScript /> > has children 1`] = `"<py-script>print(\\"Hello World!\\")</py-script>"`;

exports[`<PyScript /> > renders 1`] = `"<py-script></py-script>"`;
